lib/parser.rb

```diff
@@ -11,20 +24,49 @@ def first_line
     data.close
     content
   end
-  
+
   def generate_json
     file = @path.split("/").last
-    
-    obj = { file => {
-      :lines => count_lines()
-      } }
-      
+
+    obj = {
+      file => {
+        :lines => count_lines(),
+        :players => count_players()
+      }
+    }
+
     JSON.pretty_generate(obj)
+  end
+
+  private
+
+  def count_lines
+    File.readlines(@path).count
+  end
+
+  def count_players
+    players = []
+
+    File.readlines(@path).each do |line|
+      PATTERNS.each do |pattern|
+        player_name = line_scanner(line, pattern)
+
+        player_include(player_name, players)
+      end
     end
 
-    private
+    players
+  end
+
+  def line_scanner(line, pattern)
+    if line.include?(pattern[:word])
+      line.split(pattern[:init]).last.split(pattern[:end]).first
+    end
+  end
 
-    def count_lines
-      data = File.readlines(@path).count
+  def player_include(player_name, players)
+    if player_name
```

Comment:
Não dava para usar uma guard clause aqui?
```suggestion
    return unless player_name
    return unless players.include?(player_name)
    
    players << player_name
```

Reply:
funcionou assim:
```
  def player_include(player_name, players)
    return unless player_name
    return players << player_name unless players.include?(player_name)
  end
```